src/ServiceControl.RavenDB/RavenClientCertificate.cs

```diff
@@ -0,0 +1,51 @@
+#nullable enable
+
+namespace ServiceControl.RavenDB;
+
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+public static class RavenClientCertificate
+{
+    public static X509Certificate2? FindClientCertificate(IRavenClientCertificateInfo certInfo)
+    {
+        if (certInfo.ClientCertificateBase64 is not null)
+        {
+            try
+            {
+                var bytes = Convert.FromBase64String(certInfo.ClientCertificateBase64);
+                return new X509Certificate2(bytes, certInfo.ClientCertificatePassword);
```

Comment:
I have done some more research and found it the constructor here is obsolete. We should be switching over to the certificate loader instead. Example https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificateloader.loadpkcs12?view=net-9.0#system-security-cryptography-x509certificates-x509certificateloader-loadpkcs12(system-byte()-system-string-system-security-cryptography-x509certificates-x509keystorageflags-system-security-cryptography-x509certificates-pkcs12loaderlimits)

Reply:
The replacement APIs aren't available until .NET 9 and we're currently on .NET 8, so we'll have to do that later.